queue.go

```diff
@@ -365,6 +365,10 @@ func (queue *redisQueue) ReceiveBatch(batchSize int, waitTime time.Duration) ([]
 		// redisErrIsNil may panic because the transport may reuse connections unexpectedly.
 		// instead, perform check in-line and don't panic if Redis has the problem.
 		if err := result.Err(); err != nil {
+			// Filter these two errors out to reduce log cruft.
+			if err == redis.Nil || strings.Contains(err.Error(), "i/o timeout") {
```

Comment:
1. Thought about that. The reason the redisErrIsNil function is not used here is because of the transport reuse issue. This change is an extension of that to filter out errors associated with that issue as well.
2. Agree completely. Initial search showed this solution. Better search showed cleaner solution for go1.6.